presto-tpch/src/main/java/com/facebook/presto/tpch/StatisticsEstimator.java

```diff
@@ -0,0 +1,258 @@
+    private static final Set<TpchColumn<?>> UNSUPPORTED_COLUMNS = ImmutableSet.of(
+            CustomerColumn.ACCOUNT_BALANCE,
+            LineItemColumn.EXTENDED_PRICE,
+            OrderColumn.TOTAL_PRICE,
+            PartColumn.RETAIL_PRICE,
+            PartSupplierColumn.AVAILABLE_QUANTITY,
+            SupplierColumn.ACCOUNT_BALANCE
+    );
+
+    public TableStatisticsData estimateStats(TpchTable<?> tpchTable, Predicate<Map<TpchColumn<?>, Object>> constraint, double scaleFactor)
+    {
+        TableStatisticsData bigStatistics = readStatistics(tpchTable.getTableName(), constraint, "sf1");
+        TableStatisticsData smallStatistics = readStatistics(tpchTable.getTableName(), constraint, "tiny");
+        double rescalingFactor = smallStatistics.getRowCount() == bigStatistics.getRowCount() ? 1 : scaleFactor;
+        return new TableStatisticsData(
+                (long) (bigStatistics.getRowCount() * rescalingFactor),
+                rescale(tpchTable, bigStatistics, smallStatistics, scaleFactor)
+        );
+    }
+
+    private Map<String, ColumnStatisticsData> rescale(TpchTable<?> table, TableStatisticsData bigStatistics, TableStatisticsData smallStatistics, double scaleFactor)
+    {
+        return bigStatistics.getColumns().entrySet().stream().collect(toImmutableMap(
+                Map.Entry::getKey,
+                entry -> {
+                    String columnName = entry.getKey();
+                    TpchColumn<?> column = table.getColumn(columnName);
+                    ColumnStatisticsData bigColumnStatistics = entry.getValue();
+                    ColumnStatisticsData smallColumnStatistics = smallStatistics.getColumns().get(columnName);
+                    return rescale(column, bigColumnStatistics, smallColumnStatistics, scaleFactor);
+                }
+        ));
+    }
+
+    private ColumnStatisticsData rescale(TpchColumn<?> column, ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (UNSUPPORTED_COLUMNS.contains(column)) {
+            return ColumnStatisticsData.empty();
+        }
+        else {
+            return rescale(big, small, scaleFactor);
+        }
+    }
+
+    private ColumnStatisticsData rescale(ColumnStatisticsData big, ColumnStatisticsData small, double scaleFactor)
+    {
+        if (areClose(big, small)) {
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount(),
+                    big.getNullsCount(),
+                    checkPresent(withBoth(big.getLowestValue(), small.getLowestValue(), this::checkClose)),
+                    checkPresent(withBoth(big.getHighestValue(), small.getHighestValue(), this::checkClose))
+            );
+        }
+        else {
+            Function<Number, Double> scale = value -> value.doubleValue() * scaleFactor;
+            return new ColumnStatisticsData(
+                    big.getUniqueValuesCount().map(scale.andThen(Number::longValue)),
```

Comment:
Why is that? I have no strong preference and applied the comment, but would like to know the rationale.